pandas/_config/display.py

```diff
@@ -0,0 +1,72 @@
+"""
+Unopinionated display configuration.
+"""
+import locale
+import sys
+
+from pandas._config import config as cf
+
+# -----------------------------------------------------------------------------
+# Global formatting options
+_initial_defencoding = None
+
+
+def detect_console_encoding():
+    """
+    Try to find the most capable encoding supported by the console.
+    slightly modified from the way IPython handles the same issue.
+    """
+    global _initial_defencoding
+
+    encoding = None
```

Comment:
shouldn’t u check the global first? (if it’s set)

Reply:
The comment near the bottom suggests this is kept for debugging purposes related to matplotlib